src/shared/molecules/TypeSelector/TypeSelector.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useCallback, useRef, useState, ReactElement } from 'react';
```

Comment:
Nice to have: I think the file hierarchy is a bit weird. Instead of placing this file inside `src/shared/molecules`, it should probably be in `src/shared/components`.

The My-Data stuff could go in `src/containers/MyData` or `src/subapps/MyData`. Same for DataExplorer too.